application/frontend/controllers/MethodController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace frontend\controllers;
+
+use common\models\User;
+use frontend\components\BaseRestController;
+
+use yii\helpers\ArrayHelper;
+use yii\filters\AccessControl;
+use common\models\Method;
+
+class MethodController extends BaseRestController
+{
+    /**
+     * Access Control Filter
+     * NEEDS TO BE UPDATED FOR EVERY ACTION
+     * @return array
+     */
+    public function behaviors()
+    {
+        return ArrayHelper::merge(parent::behaviors(), [
+            'access' => [
+                'class' => AccessControl::className(),
+                'rules' => [
+                    [
+                        'allow' => true,
+                        'roles' => ['@'],
+                    ],
+                    [
+                        'allow' => true,
+                        'actions' => ['index', 'view'],
+                        'roles' => ['?'],
+                    ],
+                ]
+            ]
+        ]);
+    }
+
+    /**
+     * Return list of available reset methods for user.
+     * If user is not authenticated they should be masked.
```

Comment:
Is masking these still a todo item, or is that already being handled somewhere else?  (This looked like a `@todo` comment, but since it lacked the tag it won't be flagged in Scrutinizer.)